### ISSUE TITLE
Seperate Payment Gateway.

### ISSUE THREAD

**Issue by mohancool001** [Attendize/Attendize#179]
Hi! @daveearley . Great work on the website. I just wanted to add my own Payment gateway into the structure. I can see that the paypal express and stripe is embeded into the structure. Can you tell me what all models and database structure i have to look after if im going to ember the API calls of the payment gateway i prefer? (Assuming i can take care of routing and the controllers if i can figure out the callbackurl you need when payment gateway returns success) . Thanks !

**Comment by daveearley** [comment]
We use the Omnipay library to handle payments, so If your payment gateway is listed [here](https://github.com/thephpleague/omnipay#payment-gateways) it will be relatively easy to add it in. To add it follow the steps below. If it isn't listed you'll have to create your own extension for Omnipay and then follow the steps below.

To add a gateway first you have to add the gateway here: https://github.com/Attendize/Attendize/blob/master/database/seeds/PaymentGatewaySeeder.php

Then add the corresponding ID here: https://github.com/Attendize/Attendize/blob/master/config/attendize.php#L11 

Then add the form to fill out the gateway API credentials here:
https://github.com/Attendize/Attendize/blob/master/resources/views/ManageAccount/Partials/PaymentGatewayOptions.blade.php#L39

Then add the logic here: https://github.com/Attendize/Attendize/blob/master/app/Http/Controllers/EventCheckoutController.php#L319

In the upcoming versions it should be easier to add in your own gateway.

**Comment by mohancool001** [comment]
Oh thanks!

**Comment by jacobcat** [comment]
Hi, I followed this thread to add my own payment gateway, but there is another thing that you need to add.

There is also payment gateway specific code at https://github.com/Attendize/Attendize/blob/master/app/Http/Controllers/ManageAccountController.php#L127 that you would also need to add for your payment gateway to work.

Hope this helps.

**Comment by daveearley** [comment]
Thanks @jacobcat !

**Comment by kuldeepJain11** [comment]
HI,

Hello,

I have setup 'pesapal' payment gatway acoording to '#179'
and 'https://github.com/fruitcake/omnipay-pesapal' and also change 'is_on_site' to 1 in database, see screenshot: http://prntscr.com/jfmv6y.
I have created demo account in pesapal and received consumer key and consumer secrete and this information store against pesapal from attendize account setting: https://prnt.sc/jf7m4c.
After all this process, when i checkout from event url, getting error like this:

(index):3 Uncaught Error: You did not set a valid publishable key. Call Stripe.setPublishableKey() with your publishable key. For more info, see https://stripe.com/docs/stripe.js
at Function.b.validateKey ((index):3)
at Function.b.create ((index):2)
at Function.c.createToken ((index):2)
at HTMLFormElement. (frontend.js:4673)
at HTMLFormElement.dispatch (frontend.js:2)
at HTMLFormElement.r.handle (frontend.js:2)
b.validateKey @ (index):3
b.create @ (index):2
c.createToken @ (index):2
(anonymous) @ frontend.js:4673
dispatch @ frontend.js:2
r.handle @ frontend.js:2

can you check this?
Please guide me how can i remove this error.

Thanks

**Comment by vonec** [comment]
kuldeepJain any luck on that ?

**Comment by vonec** [comment]
resources/views/Public/ViewEvent/Embedded/EventPageCheckout.blade.php

i dont think 

@section('head')
    <script type="text/javascript" src="https://js.stripe.com/v2/"></script>
@stop

is necessary and can be conditional

**Comment by emergingdzns** [comment]
I'm having a very similar issue here. I've created a Payeezy gateway. But the payments just don't do anything when the user clicks to checkout. I was getting the stripe errors too. I removed that head part from the blade file but now I get this error in the javascript console:

```
frontend.js:4644 Uncaught ReferenceError: Stripe is not defined
    at HTMLFormElement.<anonymous> (frontend.js:4644)
    at HTMLFormElement.dispatch (frontend.js:2)
    at HTMLFormElement.r.handle (frontend.js:2)
```

The entire payment form seems to be completel coded specifically for Stripe. Payeezy is a direct API and not a tokenize thing so how do I make this thing work?
